### ISSUE TITLE
Gradle 6 deprecation warnings

### ISSUE THREAD

**Issue by boris-petrov** [johnrengelman/shadow#530]
### Shadow Version
5.2.0

### Gradle Version
6.0.1

### Expected Behavior
No warnings

### Actual Behavior
```
Property 'transformers.$0.serviceEntries' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.                                                                                                                                             
Property 'transformers.$2.data' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.                                                                                                                                                       
Property 'transformers.$3.doc' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.                                                                                                                                                        
Property 'transformers.$3.ignoreDtd' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.
```

### Gradle Build Script(s)
```gradle
task foo(type: ShadowJar) {
	from sourceSets.some.output

	configurations = [project.configurations.someRuntimeClasspath]

	archiveBaseName = 'some'
	zip64 = true

	mergeServiceFiles()
	mergeGroovyExtensionModules()
	transform(AppendingTransformer) {
		resource = 'META-INF/extensions.idx'
	}
	transform(XmlAppendingTransformer) {
		resource = 'META-INF/vfs-providers.xml'
	}

	manifest.attributes('Main-Class': 'com.company.Some')

	exclude 'META-INF/*.RSA', 'META-INF/*.SF','META-INF/*.DSA'

	duplicatesStrategy = DuplicatesStrategy.FAIL
}
```

**Comment by iainmcgin** [comment]
@johnrengelman I still see a warning related to the `serviceEntries` property in [ServiceFileTransformer](https://github.com/johnrengelman/shadow/blob/a7f0908521029a848b5de2e2a131af146bd213cd/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.groovy#L54) when running a build with 5.2.0 of the plugin:

```
Property 'transformers.$0.serviceEntries' is not annotated with an input or output annotation.
```

I don't know much about Gradle internals, but from digging around the documentation I think this property should be annotated as `@Input`.

**Comment by xalvarez** [comment]
@iainmcgin bear in mind that v5.2.0 was released last November so it doesn't include the changes above.

**Comment by iainmcgin** [comment]
What changes are you referring to? The closest I could find that I think are relevant are in #524, which appear to have been included in 5.2.0. I'm happy to wait regardless, as this doesn't break anything, just curious now that I'm looking to know what PRs are involved in fixing this issue as they aren't directly linked here.

**Comment by johnrengelman** [comment]
This might not be addressed. My local builds of shadow itself do not output these warnings but I’m guessing that’s because these transformers are not used. I’ll need to figure out a way to check all the transformer implementations during the build. 

Any fix for this will be in 6.0.0 (unreleased) per the milestone in this issue.

**Comment by ostrya** [comment]
I see similar warnings still (e.g. for ApacheNoticeResourceTransformer). Should I file a new issue, or are you already tracking this?
